qml/pages/ProgramsPage.qml

```diff
@@ -7,62 +7,69 @@ Page {
     id: page
 
     property var category: ({})
+    property var series: ({})
+
     property int offset: 0
     property int limit: 25
+    property bool programsEnd: false
 
     function getPrograms() {
-        YleApi.getProgramsByCategoryId(category.id, limit, offset)
-            .then(function(programs) {
-                console.log('programs', programs)
-                listView.model = programs
-            })
-            .catch(function(error) {
-                console.log('error', error)
-                listView.model = []
-            })
+        if (category.id) {
+            YleApi.getProgramsByCategoryId(category.id, limit, offset)
+                .then(function(programs) {
```

Comment:
extract this function since it is duplicated here and in the alternative branch?

Reply:
Good point, I'll fix that

Reply:
:+1: